src/Extension/EscaperExtension.php

```diff
@@ -100,6 +104,41 @@ public function getName()
     {
         return 'escaper';
     }
+
+    /**
+     * @internal
+     */
+    public function getSafeClasses(): array
+    {
+        self::$safeClassesFrozen = true;
+
+        return $this->safeClasses;
+    }
+
+    public function setSafeClasses(array $safeClasses = []): void
+    {
+        if (self::$safeClassesFrozen) {
+            throw new \LogicException('Unable to set the safe classes for escaping as twig_escape_filter() has already been initialized.');
```

Comment:
wouldn't this cause issues when running tests of a Symfony project where the kernel could be booted lots of time in the same process ? The first boot will instantiate Twig and freeze safe classes at some point. The next unit test will boot a kernel, which will then create another Twig instance trying to set the same safe classes, but this will break here. Forcing to use process isolation for the whole testsuite as soon as the project uses safe classes is a bad idea.
And the same will happen for phppm users btw, but they cannot use process isolation (as that would defeat the whole point).

Reply:
Fixed